docs/jwst/introduction.rst

```diff
@@ -196,6 +196,81 @@ then supply the cfg file as a keyword argument:
  Detector1Pipeline.call('jw00017001001_01101_00001_nrca1_uncal.fits', config_file='calwebb_detector1.cfg')
 
 
+.. _intro_file_conventions:
+
+Input and Output File Conventions
+=================================
+
+.. _intro_input_file_discussion:
+
+Input Files
+-----------
+
+There are two general types of input to any step or pipeline: references files
+and data files.  The references files, unless explicitly
+overridden, are provided through CRDS.
+
+Data files are the science input, such as exposure FITS files and association
+files. All data is assumed to be co-resident in the directory where the primary
+input file is located. This is particularly important for associations: JWST
+associations contain file names only. All files referred to by an association
+are expected to be located in the directory the association file is located.
+
+.. _intro_output_file_discussion:
+
+Output Files
+------------
+
+Output files will be created either in the current working directory, or where
+specified by the :ref:`output_dir <intro_output_directory>` configuration
+parameter.
+
+File names for the outputs from pipelines and steps come from
+three different sources:
+
+- The name of the input file
+- The product name defined in an association
+- As specified by the :ref:`output_file <intro_output_file>` argument
+
+Regardless of the source, each pipeline/step uses the name as a "base
+name", on to which several different suffixes are appended, which
+indicate the type of data in that particular file. A list of the main suffixes
+can be :ref:`found below <pipeline_step_suffix_definitions>`.
+
+Output File and Associations
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Stage 2 pipelines can take an individual file or an
+:ref:`association <associations>` as input. Nearly all Stage 3
+pipelines require an associaiton as input. Normally, the output file
+is defined in each association's `product_name`.
+
+If there is need to produce multiple versions of a calibration based
+on an association, it is highly suggested to use `output_dir` to place
+the results in a different directory instead of using `output_file` to
+rename the output files. Stage 2 pipelines do not allow the override
```

Comment:
👎

Reply:
"Stage 2 pipelines do not allow the override ..." Really? Is that because "output_file" can only specify a single name, but the stage 2 pipelines create multiple products? So "output_file" is expected to be a complete file name, as opposed to just a base name?

Reply:
The reason is because they handle multiple products, and what setting `--output_file` in that situation means is not clear. So currently disabled. This situation does need to be reconsidered, such as in #1096 and #2159.